### ISSUE TITLE
When user is at point "A", OsmAnd is unable to chart a "B to C" or "B to A" routes.

### ISSUE THREAD

**Issue by raindropsfromsky** [osmandapp/Osmand#4928]
Reporting for OsmAnd~ 3.0.0#23555m running on OnePlus 3T with Android 8.8.8 (OxygenOS 5.0.1).

Suppose the user is at point A. In that case, OsmAnd cannot find clean routes:
1. From point B to Point C
2. From point B to point A 

See the examples below.

**Reverse journey (Point B -> Point A):**
OsmAnd draws a quick straight line from point A to point B, and then charts the actual route.
![image](https://user-images.githubusercontent.com/9047168/35101080-305d51ac-fc84-11e7-8761-54e6b63eeb73.png)

**Journey between unrelated points (point B -> Point C):**
![image](https://user-images.githubusercontent.com/9047168/35101092-3a2530c4-fc84-11e7-9e51-f99061a0bca1.png)

**Comment by sonora** [comment]
Narayan, it looks like you had hit the "GO" button in your cases, meaning you want to **start following a turn-by-turn navigation**, which of course always relates to your actual position.

To do what you need, just remain on the route planning screen, i.e. where you input the start point and destinations. Staying on that screen without starting the voice navigation will show you the route you request.

**Comment by raindropsfromsky** [comment]
Yes, confirmed: I did press "go".

But then reaching point B should also be via the available road network, following all traffic rules. (not "as the crow flies").  This journey should be either real or imaginary; not a mix of both!

**Comment by sonora** [comment]
I cannot actually reproduce your observation. For me it works like it has been for years: Calculate a route to anywhere with a start (From:) point. If you then tap GO, the start point is simply replaced by your actual position, and the route is correctly recalculated. Not sure at the moment what you do differently...

**Comment by raindropsfromsky** [comment]
It happens always in my case.
Simplest case: 
1. Configure two spots as your favorites.
2. Press the **Navigate** button
3. In the **From** and **To** fields, select the favorites.
4. OsmAnd creates the route (so far, normal behavior)
5. Now press "Go". This is where OsmAnd creates a straight line section ("as the crow flies")

**Comment by sonora** [comment]
Strange, that procedure works ok for me. But I just notice that in both of your screenshots your own position marker is gray, not yet blue, looks like there is no GPS fix yet. So I guess your issue is simply that you are still waiting for the GPS fix?

**Comment by raindropsfromsky** [comment]
I repeated the process with GPS available, but that didn't change the
behaviour.

**Comment by sonora** [comment]
There cannot be navigation without live position updates. :) 

But I do not know what more to say - I simply cannot reproduce (testing with latest nightly). The only setting I know of which has vaguely to do with behavior like this is "Let OsmAnd calculate the first and last route segment" (should be effective only for GPX routing, but you never know). Maybe to make sure select just any GPX file for routing, then make sure that setting is enabled, then quit the route again and try your experiment again. 

Other settings I have as follows, but should not really have an impact: Disable complex routing is OFF, and Smart route recalculation is ON. (Both under Development plugin). Snap to road I always have OFF, but that is a matter of taste.

**Comment by FTABoy** [comment]
i have the same problem.

**Comment by sonora** [comment]
I guess someone would have to post version and settings info to make some leeway  here.

**Comment by raindropsfromsky** [comment]
I posted the version. It continues in the latest version as of today also.

**Comment by FTABoy** [comment]
@sonora 
version 2.8.2
to reproduce this problem try to lock current position of gps on map and after signal lose (or disabling gps on device) which makes the position marker gray try to navigate between two favorite location. this will cause the straight line... no special setting is applied.
![photo_2018-01-21_16-39-15](https://user-images.githubusercontent.com/25687276/35194501-bbc0820e-fec9-11e7-818e-f61af7060e3f.jpg)

**Comment by raindropsfromsky** [comment]
Yes, this intermittent GPS availability may be a clue. But then Osmand  is
not supposed to need the live GPS bearing to calculate the route. And
besides, it has already found the current location, and included it in the
route. So even if the GPS is lost, it should not matter.

**Comment by oxxm** [comment]
Can confirm a similar behaviour with activated GPS. 

Just calculate route from point A to point B on the map. The actual GPS position is taken into consideration and is part of the calculated route, although it should be just ignored if it is not set as starting point.

**Comment by sonora** [comment]
Related: #22595.
